paimon-spark/paimon-spark-common/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/PaimonSparkSqlExtensionsParser.scala

```diff
@@ -0,0 +1,268 @@
+      parse(sqlTextAfterSubstitution)(parser => astBuilder.visit(parser.singleStatement()))
+        .asInstanceOf[LogicalPlan]
+    } else {
+      delegate.parsePlan(sqlText)
+    }
+  }
+
+  /** Parses a string to an Expression. */
+  override def parseExpression(sqlText: String): Expression = {
+    delegate.parseExpression(sqlText)
+  }
+
+  /** Parses a string to a TableIdentifier. */
+  override def parseTableIdentifier(sqlText: String): TableIdentifier = {
+    delegate.parseTableIdentifier(sqlText)
+  }
+
+  /** Parses a string to a FunctionIdentifier. */
+  override def parseFunctionIdentifier(sqlText: String): FunctionIdentifier = {
+    delegate.parseFunctionIdentifier(sqlText)
+  }
+
+  /**
+   * Creates StructType for a given SQL string, which is a comma separated list of field definitions
+   * which will preserve the correct Hive metadata.
+   */
+  override def parseTableSchema(sqlText: String): StructType = {
+    delegate.parseTableSchema(sqlText)
+  }
+
+  /** Parses a string to a DataType. */
+  override def parseDataType(sqlText: String): DataType = {
+    delegate.parseDataType(sqlText)
+  }
+
+  /** Parses a string to a multi-part identifier. */
+  override def parseMultipartIdentifier(sqlText: String): Seq[String] = {
+    delegate.parseMultipartIdentifier(sqlText)
+  }
+
+  /** Returns whether SQL text is command. */
+  private def isCommand(sqlText: String): Boolean = {
+    val normalized = sqlText
+      .toLowerCase(Locale.ROOT)
+      .trim()
+      .replaceAll("--.*?\\n", " ")
+      .replaceAll("\\s+", " ")
+      .replaceAll("/\\*.*?\\*/", " ")
+      .trim()
+    normalized.startsWith("call")
+  }
+
+  protected def parse[T](command: String)(toResult: PaimonSqlExtensionsParser => T): T = {
+    val lexer = new PaimonSqlExtensionsLexer(
+      new UpperCaseCharStream(CharStreams.fromString(command)))
+    lexer.removeErrorListeners()
+    lexer.addErrorListener(PaimonParseErrorListener)
+
+    val tokenStream = new CommonTokenStream(lexer)
+    val parser = new PaimonSqlExtensionsParser(tokenStream)
+    parser.addParseListener(PaimonSqlExtensionsPostProcessor)
+    parser.removeErrorListeners()
+    parser.addErrorListener(PaimonParseErrorListener)
+
+    try {
+      try {
+        parser.getInterpreter.setPredictionMode(PredictionMode.SLL)
+        toResult(parser)
+      } catch {
+        case _: ParseCancellationException =>
+          tokenStream.seek(0)
+          parser.reset()
+          parser.getInterpreter.setPredictionMode(PredictionMode.LL)
+          toResult(parser)
+      }
+    } catch {
+      case e: PaimonParseException if e.command.isDefined =>
+        throw e
+      case e: PaimonParseException =>
+        throw e.withCommand(command)
+      case e: AnalysisException =>
+        val position = Origin(e.line, e.startPosition)
+        throw new PaimonParseException(Option(command), e.message, position, position)
+    }
+  }
+
+  override def parseQuery(sqlText: String): LogicalPlan = {
```

Comment:
I think we can delete `PaimonSparkSqlExtensionsParser` files in Spark3.1, 3.2, 3.2, just remain this in spark-common.
And remove this `override` identifier from this method that added in https://issues.apache.org/jira/browse/SPARK-37266.